src/main/index.js

```diff
@@ -68,14 +73,56 @@ if (appConfig.appType === 'development') {
 	}
 }
 
+// NOTE: Has to be set up after user data directory is updated
 const configStore = createConfigStore()
 
+let sentryUser = configStore.get('sentryUser')
+
+// NOTE: The retrieved value may be the default based on config store schema,
+// which is not immediately persisted upon initialization.
+// We want to make sure that this value is persisted upon startup, even if it might be
+// rotated shortly after.
+configStore.set('sentryUser', sentryUser)
+
+if (shouldRotateSentryUser(sentryUser)) {
+	log('Rotating Sentry user')
+	const newSentryUser = generateSentryUser()
+	configStore.set('sentryUser', newSentryUser)
+	sentryUser = newSentryUser
+}
+
+/** @type {import('../shared/app.js').SentryEnvironment} */
+let sentryEnvironment = 'development'
+
+if (appConfig.appType === 'release-candidate') {
+	sentryEnvironment = 'qa'
+} else if (appConfig.appType === 'production') {
+	sentryEnvironment = 'production'
+}
+
+// NOTE: Has to be set up after user data directory is updated
+// https://docs.sentry.io/platforms/javascript/guides/electron/#app-userdata-directory
+Sentry.init({
+	dsn: 'https://f7336c12cc39fb0367886e31036a6cd7@o4507148235702272.ingest.us.sentry.io/4509803831820288',
+	// TODO: Only works on app startup. Any changes to `diagnosticsEnabled` while the app is running will not
+	// take effect here until the app is restarted.
```

Comment:
Nitpick:  ~`TODO`~ -> `NB` - I think it's probably going to be fine to only do this on restart, especially on desktop.

Reply:
TIL: `NB`. I've been using `NOTE` so will use that for consistency